Reduce noise in logs when running tests

msprunck opened the pull request:
> Related https://github.com/threatgrid/ctia/pull/1189
> Related https://github.com/threatgrid/ctia/pull/1190#issuecomment-1025980043

Adds a new configuration file for logback that change the configuration during the tests. By default, it disables logs in the root logger and only allows logger that start with `ctia`.

During its initialization, logback tries to find a file logback-test.xml in the classpath. If no such file is found, it checks for the file logback.xml in the classpath.

This PR adds a `logback-test.xml` file which is only available in the test resource folder. It will not be deployed by `uberjar` or used with the command `lein run -m ctia.main`.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Disable logs during tests
```

yogsototh reviewed (approved):
LGTM!

frenchy64 commented:
@msprunck I think there's a problem with the classpath, the CI is still noisy.

I think we need to move `test/resources` into its own folder `test-resources` and then add `:resource-paths ["test-resources"]` to the `:test-impl` alias.

I think `test/resources` being under `test` is a mistake.

msprunck commented:
> @msprunck I think there's a problem with the classpath, the CI is still noisy.
The goal of this PR is to retrieve the same level of logs we had before I added the configuration file `logback.xml`. It provides a new configuration file `logback-test.xml` which is specific to the test profile.

I tried to disable logs during tests with the `OFF` level or reduce them with `ERROR` or `WARN` level but it produces: 
- race conditions in the migration tests.
- errors in a [test](https://github.com/threatgrid/ctia/blob/6a08126da103c7ec54aa4c967c374664e832cea8/test/ctia/logger_test.clj#L47) that depends on INFO log level.
 
> I think we need to move `test/resources` into its own folder `test-resources` and then add `:resource-paths ["test-resources"]` to the `:test-impl` alias.
Why `:test-impl` and not `test`?
 
> I think `test/resources` being under `test` is a mistake
You are right, I'm not confortable with this folder structure. I changed it.

msprunck commented on project.clj:
```diff
@@ -204,12 +203,11 @@
                        :main ctia.main
                        :uberjar-name "ctia.jar"
                        :uberjar-exclusions [#"ctia\.properties"]}
-             :test {:jvm-opts ["-Dlog.console.threshold=WARN"]
```
This property is specific to log4j which is no longer used

ereteog reviewed (approved):
LGTM

frenchy64 commented:
> Why :test-impl and not test?

Whoops, never mind. Might be worth leaving it in `:dev` too https://github.com/technomancy/leiningen/blob/master/doc/PROFILES.md#default-profiles

> I tried to disable logs during tests with the OFF level or reduce them with ERROR or WARN levels but it produces...

Ah ok. I think the INFO logs make the output about 5-10x longer than before the `logback.xml` file, that's why I thought there was a classpath issue. Thanks for clarifying!

msprunck commented:
> Whoops, never mind. Might be worth leaving it in `:dev` too https://github.com/technomancy/leiningen/blob/master/doc/PROFILES.md#default-profiles

I don't think so. If we leave it in the `:dev` profile, logback will use `logback-test.xml` when running CTIA locally. Usually `logback-test.xml` is dedicated for tests. You can change the log configuration with `logback.xml`when running CTIA locally with `logback.xml`

agzam reviewed (approved):
nice!
